### PR TITLE
fix: recognize classes pulled in via @use/@forward/@import (#90)

### LINKED ISSUE #90: SCSS: classes pulled in via @use are reported as non-existent
Checked on version 0.5.2

## Example

`src/_shared.scss`:

```scss
.box {
  color: red;
}
```

`src/Foo.module.scss`:

```scss
@use 'shared';
```

`src/Foo.tsx`:

```tsx
import styles from './Foo.module.scss'

export const Foo = () => <div className={styles.box} />
```

Running `check-unused-css src`:

```
Found 1 classes used in source files but non-existent in CSS:
  Foo.tsx:3:49 - .box
```

### PR DESCRIPTION
**Pull request by malinindev**
## Problem

Fixes #90. When a SCSS module pulls in a partial via `@use`, `@forward` or the legacy `@import`, the partial's top-level rules are emitted into the module's compiled CSS — so those classes are real, usable classes. The tool only parsed the module file itself, so it falsely reported imported classes as **non-existent**.

```scss
/* _shared.scss */ .box { color: red; }
/* Foo.module.scss */ @use 'shared';
```
```tsx
import styles from './Foo.module.scss'
export const Foo = () => <div className={styles.box} />  // .box was wrongly flagged
```

## Fix

New `scssImports` utility under `getUnusedClassesFromCss/utils/`:

- **`extractScssImportPaths`** — pulls module specs out of `@use`/`@forward`/`@import`. Plain-CSS `@import`s are skipped per Sass's own rule (`.css` extension, `url(...)`, `http(s)://`, or a trailing media query).
- **`resolveScssImport`** — resolves a spec to a file using Sass load rules: `_name` partials, `.scss/.sass/.css` extensions, and directory `_index`.
- **`collectPartialClasses`** — walks the import graph transitively (BFS) with a visited-set guard against cycles (legal with `@import`) and diamonds. Non-Sass files are skipped by extension.

Wired into both branches:
- **non-existent**: partial classes count as existing.
- **unused**: partial classes are seeded as used, so a local class sharing a name isn't flagged. Classes that exist **only** in a partial are never unused candidates — the importing module doesn't own them.

## Tests

- E2E (`scssAtRuleImports.test.ts`): `@use`, namespaced `@use`, `@forward` chained through `_index`, legacy `@import`, subdirectory partials, diamond dedup, `@import` cycle termination, missing partial, plus two guards that a genuinely missing class and a genuinely unused local class are still reported.
- Unit tests for `extractScssImportPaths`, `resolveScssImport`, `collectPartialClasses` covering parsing edge cases (comments, multi-line `with`, `as *`, media queries) and resolution/recursion edge cases.

All 984 tests pass; `check:all` (format, lint, types, tests) is green.

### CODE REVIEW

File: src/lib/getUnusedClassesFromCss/utils/scssImports/collectPartialClasses.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    } catch {
+      continue;
+    }
+
+    for (const className of extractCssClasses(partialContent)) {
+      classNames.add(className);
+    }
+
+    enqueueImportsFrom(partialContent, path.dirname(partialPath));
+  }
```
[the remark this answers is not in the document]
Comment (malinindev) [reply src/lib/getUnusedClassesFromCss/utils/scssImports/collectPartialClasses.ts]: Done. Wrapped extractCssClasses in try/catch so a malformed partial is skipped, and switched the queue to an index-based walk. Added a test.

File: src/lib/getNonExistentClassesFromCss.ts [hunk 2]
```diff
@@ -27,7 +29,12 @@ export const getNonExistentClassesFromCss = async ({
   }
 
   const cssContent = getContentOfFiles({ files: [cssFile], srcDir });
-  const cssClasses = extractCssClasses(cssContent);
+
+  // Classes are real if they are defined in this file OR pulled into it via
+  // `@use`/`@forward`/`@import` (those emit the partial's rules into the
+  // module's compiled CSS), so both sets count as existing (issue #90).
+  const partialClasses = collectPartialClasses(path.join(srcDir, cssFile));
+  const cssClasses = [...extractCssClasses(cssContent), ...partialClasses];
```
[the remark this answers is not in the document]
Comment (malinindev) [reply src/lib/getNonExistentClassesFromCss.ts:42]: Done. cssClasses is now a Set and the lookup uses .has().

File: src/lib/getUnusedClassesFromCss/utils/scssImports/collectPartialClasses.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+  const queue: string[] = [];
+  const enqueueImportsFrom = (content: string, fromDir: string): void => {
+    for (const spec of extractScssImportPaths(content)) {
+      const resolved = resolveScssImport(spec, fromDir);
+      if (resolved && !visited.has(resolved)) {
+        visited.add(resolved);
+        queue.push(resolved);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (malinindev) [reply src/lib/getUnusedClassesFromCss/utils/scssImports/collectPartialClasses.ts:55]: Not filtering by extension here on purpose. Sass loads code.css for a bare @use 'code' / @import 'code' and inlines its rules, so those classes are real and must be collected. Filtering to .scss/.sass would drop them. A spec with the .css extension is a plain CSS import and is already filtered upstream. Clarified the docstring and added a test for the bare-spec .css case.
